src/modules/navigator/mission.cpp

```diff
@@ -281,6 +282,8 @@ Mission::on_active()
 		replayCachedTriggerItems();
 	}
 
+	replayCachedSpeedChangeItems();
```

Comment:
so this one
1. updates to the latest potential speed change ONLY by the mission item iteration
2. and then sends the vehicle command such that
3. `cruising_speed_sp_update()` picks up that command when it checks for do set speed commands? (where it would also pick up any operator freeform do set speed commands?)

Reply:
>ONLY by the mission item iteration
Can you expand what you mean?

But yes otherwise that's what happens. Note that also the fixed-wing position controller and the multicopter flight mode manager listen to VEHICLE_CMD_DO_CHANGE_SPEED and execute it independently of `cruising_speed_sp_update()` (which is only run in Mission, but not in eg Loiter).

Reply:
Maybe we could even remove `cruising_speed_sp_update()`? It's a fundamental question if the speed setpoint should be part of the position setpoint or if it's rather an independent system setting.

Reply:
Removed it in 4948bb70645e2aa53c1f5b74aea70f8ea746a4ff. I feel like we should rather de-couple path handling (through the position setpoints) and speed setpoints (independent of the path) for now.